# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1321868325 0
#      Mon Nov 21 09:38:45 2011 +0000
# Branch gui
# Node ID 36f90899e0581d96cf2a5ee912d6a28138736296
# Parent  22e16fd68b8a4b161d45132b770b91373ee7e7e7
[Win32] For compilation on MinGW due to wchar_t* conversion.

* src/terminal/win32/QConsole.cpp (QConsolePrivate::QConsolePrivate): Call
QString::fromWCharArray instead of QString::fromUtf16.
(QConsolePrivate::monitorConsole): Likewise.

diff --git a/gui/src/terminal/win32/QConsole.cpp b/gui/src/terminal/win32/QConsole.cpp
--- a/gui/src/terminal/win32/QConsole.cpp
+++ b/gui/src/terminal/win32/QConsole.cpp
@@ -212,17 +212,17 @@ QConsolePrivate::QConsolePrivate (QConso
        m_bufferSize.height ());
   log ("  window: (%d, %d) -> (%d, %d) [%d x %d]\n",
        m_consoleRect.left (), m_consoleRect.top (),
        m_consoleRect.right (), m_consoleRect.bottom (),
        m_consoleRect.width (), m_consoleRect.height ());
 
   wchar_t titleBuf[260];
   GetConsoleTitleW (titleBuf, sizeof (titleBuf));
-  q->setWindowTitle (QString::fromUtf16 (titleBuf));
+  q->setWindowTitle (QString::fromWCharArray (titleBuf));
 
   m_font.setFamily ("Lucida Console");
   m_font.setPointSize (9);
   m_font.setStyleHint (QFont::TypeWriter);
   m_backgroundColor = Qt::black;
 
   m_buffer = m_tmpBuffer = 0;
 
@@ -525,17 +525,17 @@ void QConsolePrivate::updateConsoleView 
 void QConsolePrivate::monitorConsole (void)
 {
   CONSOLE_SCREEN_BUFFER_INFO sbi;
   HANDLE hStdOut = GetStdHandle (STD_OUTPUT_HANDLE);
 
   static wchar_t titleBuf[260];
 
   GetConsoleTitleW (titleBuf, sizeof (titleBuf));
-  QString title = QString::fromUtf16 (titleBuf);
+  QString title = QString::fromWCharArray (titleBuf);
 
   if (title != m_title)
     {
       q->setWindowTitle (title);
       emit q->titleChanged (title);
     }
 
   if (GetConsoleScreenBufferInfo (hStdOut, &sbi))
